docs: update memory leak debugging skill (#2330)
ChromeDevTools/chrome-devtools-mcp: skills/memory-leak-debugging/SKILL.md

---
name: memory-leak-debugging
description: Diagnoses and resolves memory leaks in JavaScript/Node.js applications. Use when a user reports high memory usage, OOM errors, or wants to capture, compare, or inspect heap snapshots with Chrome DevTools MCP memory tools.
---

# Memory Leak Debugging

This skill provides expert guidance and workflows for finding, diagnosing, and fixing memory leaks in JavaScript and Node.js applications using Chrome DevTools MCP.

## Core Principles

- **Prefer MCP memory tools:** Do NOT attempt to read raw `.heapsnapshot` files directly, as they are extremely large and will consume too many tokens. Use the Chrome DevTools MCP heap snapshot tools to summarize, compare, and inspect snapshots.
- **Isolate the Leak:** Determine if the leak is in the browser (client-side) or Node.js (server-side).
- **Common Culprits:** Look for detached DOM nodes, unhandled closures, global variables, event listeners not being removed, and caches growing unbounded. _Note: Detached DOM nodes are sometimes intentional caches; always ask the user before nulling them._

> - **Close Loaded Snapshots:** Heap snapshots can be large. Use `close_heapsnapshot` after inspecting a file to release memory held by the MCP server.

Reviewer: ```suggestion
- **Close Loaded Snapshots:** After completing you full investigation use `close_heapsnapshot` to free up memory on the MCP server.
```

I think this may be a little bit better as the first one can be interpreted as release after every tool call, which will effectively re-parse a heapsnapshot.

Author: Updated in `71dc4f9` to make the cleanup step explicit after heap snapshot investigation.

Validation:
- `git diff --check origin/main...HEAD`
- `npm run check-format`
